src/cockpit/channels/metrics.py

```diff
@@ -149,6 +152,9 @@ def send_updates(self, samples: Dict[str, Any], last_samples: Dict[str, Any]):
                 else:
                     data.append([val for val in value.values()])
             else:
+                assert not isinstance(value, dict)
+                assert not isinstance(old_value, dict)
```

Comment:
This breaks the test and this is SUPER fun,

So accessing `last_samples[metricinfo.desc.name]` causes it to be set to the default value a dict. Which means it is always set to `{}` if it's not initialised yet.